### DISCUSSION TITLE
The v-if condition does not work

### DISCUSSION THREAD

**Discussion by miladfa7** [dm4t2/vue-currency-input#301] [Q&A]
i use `vue currency Input` for input formatted numbers.
I created two files vuejs 3 :

**CurrencyInput.vue** 
```
<template>
  <input  class="form-control form-control-solid" ref="inputRef" type="text" :value="formattedValue" />
</template>

<script>
import { useCurrencyInput } from "vue-currency-input";
import { watch } from 'vue'

export default {
  name: "CurrencyInput",
  components:{
    
  },
  props: {
    modelValue: Number,
    options: Object,
  },
  setup(props) {
    const {  inputRef, formattedValue, setOptions, setValue } = useCurrencyInput(props.options);
    watch(
      () => props.modelValue,
      (value) => {
        // Vue 2: props.value
        setValue(value)
      }
    )
    watch(
      () => props.options,
      (options) => {
        setOptions(options)
      }
    )
    return { inputRef, formattedValue };
  },
};
</script>
```
**I wrote this part briefly**
**Project.vue**
```
<template>
       <botton @click="showPrice()">add</botton>
       <div v-if="show==true">
        <CurrencyInput v-model.lazy="project.price" :options=options />
        </div>
</template>

<script>
  import CurrencyInput from "./CurrencyInput.vue";

export default {
  name: "project",
  components:{
    CurrencyInput
  },
  setup() {
   
   const showPrice = async () => {
       show = true
   
}

</script>
```

**The problem is that when clicked on _add_ the if condition does not work and is not displayed** (CurrencyInput v-model.lazy="project.price" :options=options )

notice: when I use **input** instead **CurrencyInput** in Project.vue, the code works correctly.

**Comment by miladfa7** [comment]
I used this **CurrencyInput** in two components that are in the same path, I think I should **stop or async watcher**?
Do you think i can use **async** in watcher?

**Comment by miladfa7** [comment]
@dm4t2 
please help me

**Comment by miladfa7** [comment]
when i used watch and async v-model in two component that same path , this gives the error

**`Uncaught (in promise) TypeError: numberInput is undefined`** 

Please Help me,

**Comment by dm4t2** [comment]
Your issue is not clear. Can you provide a minimal reproduction, for example a [CodeSandbox](https://codesandbox.io/s/vue)?
